entc/integration/template/ent/template/query.tmpl

```diff
@@ -0,0 +1,21 @@
+{{/*
+Copyright 2019-present Facebook Inc. All rights reserved.
+This source code is licensed under the Apache 2.0 license found
+in the LICENSE file in the root directory of this source tree.
+*/}}
+
+{{/* The line below tells Intellij/GoLand to enable the autocompletion based *gen.Graph type. */}}
+{{/* gotype: entgo.io/ent/entc/gen.Graph */}}
+
+{{ define "dialect/sql/query/fields" }}
+	{{- with $.UnexportedForeignKeys }}
+		withFKs bool
+	{{- end }}
+	// additional query fields.
+	extra string
+{{ end }}
+
+{{ define "client/query/init" }}
```

Comment:
not a fan of init, maybe "client/query/new"

Reply:
See my comment above, I think it will look better when you define the fields in one template name `"dialect/sql/query/fields/additional/example"`, and initialize them in another template named `"dialect/sql/query/fields/init/example"`.